### PR TITLE
Feature/revsdl 1954 same id name

### PR DESCRIPTION
**Pull request by bonewell**
SDL must allow registration of same-named REMOTE_CONTROL applications with same appIDs (amend existing SDL behavior of "DUBLICATE_NAME" and "DISALLOWED" results in the mentioned case).

### CODE REVIEW

File: src/components/application_manager/src/commands/mobile/register_app_interface_request.cc [hunk 1]
```diff
@@ -130,12 +131,163 @@ struct IsSameNickname {
...
+
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Commands")
 
 namespace application_manager {
 
 namespace commands {
 
+#ifdef SDL_REMOTE_CONTROL
+struct IsSameAppId {
```
Comment (VladSemenyuk) [inline src/components/application_manager/src/commands/mobile/register_app_interface_request.cc]: Bad name for struct

Comment (bonewell) [inline src/components/application_manager/src/commands/mobile/register_app_interface_request.cc]: It is functor. It is good to use verb for functor.

File: src/components/application_manager/src/commands/mobile/register_app_interface_request.cc [hunk 2]
```diff
@@ -130,12 +131,163 @@ struct IsSameNickname {
...
+    bool other_is_driver = other->device() == policy::PolicyHandler::instance()->PrimaryDevice();
+    return is_driver_ == other_is_driver;
+  }
+  const std::string& app_id_;
+  bool is_remote_control_;
+  bool is_driver_;
+};
+struct IsSameAppName {
+  IsSameAppName(const std::string& name,
+                         const smart_objects::SmartArray *vr_synonyms,
```
Comment (bonewell) [inline src/components/application_manager/src/commands/mobile/register_app_interface_request.cc]: wrong indent

File: src/components/policy/src/policy/src/policy_manager_impl.cc [hunk 3]
```diff
@@ -879,6 +879,26 @@ bool PolicyManagerImpl::IsPredataPolicy(const std::string &policy_app_id) {
...
+  int operator() (const policy_table::AppHMITypes::value_type item) {
+    return policy_table::AppHMIType(item);
+  }
+};
+
+bool PolicyManagerImpl::GetHMITypes(const std::string& application_id,
+                                     std::vector<int> *app_types) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (cache_->IsDefaultPolicy(application_id)) {
+    return false;
```
Comment (bonewell) [inline src/components/policy/src/policy/src/policy_manager_impl.cc]: false only for remote control application

**Pull request merged**